From 61322cb18dc3d5c96fd4503c978fb8a9217f84fd Mon Sep 17 00:00:00 2001
From: Adrian Freihofer <adrian.freihofer@scs.ch>
Date: Wed, 14 Feb 2018 21:56:54 +0100
Subject: [PATCH 12/15] Install iiod SysVinit script

If WITH_SYSVINIT is set to ON the init script of iiod gets
installed into SYSVINIT_INSTALL_DIR folder.
SYSVINIT_INSTALL_DIR defaults to /etc/init.d.

Signed-off-by: Adrian Freihofer <adrian.freihofer@gmail.com>
---
 CMakeLists.txt                             | 3 +++
 iiod/CMakeLists.txt                        | 7 +++++++
 iiod/init/{iiod.init => iiod.init.cmakein} | 6 +++---
 3 files changed, 13 insertions(+), 3 deletions(-)
 rename iiod/init/{iiod.init => iiod.init.cmakein} (77%)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 75852a6..32a22ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -408,6 +408,9 @@ if(WITH_IIOD)
 	option(WITH_SYSTEMD "Enable installation of systemd service file for iiod" OFF)
 	set(SYSTEMD_UNIT_INSTALL_DIR /lib/systemd/system CACHE PATH "default install path for systemd unit files")
 
+	option(WITH_SYSVINIT "Enable installation of SysVinit script for iiod" OFF)
+	set(SYSVINIT_INSTALL_DIR /etc/init.d CACHE PATH "default install path for SysVinit scripts")
+
 	if (NOT PTHREAD_LIBRARIES)
 		message(WARNING "IIOD requires threads support; disabling")
 		set(WITH_IIOD OFF CACHE BOOL "" FORCE)
diff --git a/iiod/CMakeLists.txt b/iiod/CMakeLists.txt
index 76e14fa..8947601 100644
--- a/iiod/CMakeLists.txt
+++ b/iiod/CMakeLists.txt
@@ -67,3 +67,10 @@ if (WITH_SYSTEMD)
 	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/init/iiod.service.cmakein ${PROJECT_BINARY_DIR}/init/iiod.service)
 	install(FILES ${PROJECT_BINARY_DIR}/init/iiod.service DESTINATION ${SYSTEMD_UNIT_INSTALL_DIR})
 endif()
+
+if (WITH_SYSVINIT)
+	configure_file(${CMAKE_CURRENT_SOURCE_DIR}/init/iiod.init.cmakein ${PROJECT_BINARY_DIR}/init/iiod)
+	install(FILES ${PROJECT_BINARY_DIR}/init/iiod
+	        PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ
+	        DESTINATION ${SYSVINIT_INSTALL_DIR})
+endif()
diff --git a/iiod/init/iiod.init b/iiod/init/iiod.init.cmakein
similarity index 77%
rename from iiod/init/iiod.init
rename to iiod/init/iiod.init.cmakein
index d21ed00..2799925 100644
--- a/iiod/init/iiod.init
+++ b/iiod/init/iiod.init.cmakein
@@ -23,7 +23,7 @@ fi
 case "$1" in
 	start)
 		log_daemon_msg "Starting IIO Daemon" "iiod" || true
-		if start-stop-daemon -S -b -q -m -p /var/run/iiod.pid -x /usr/sbin/iiod -- $IIOD_OPTS; then
+		if start-stop-daemon -S -b -q -m -p /var/run/iiod.pid -x @CMAKE_INSTALL_FULL_SBINDIR@/iiod -- $IIOD_OPTS; then
 			log_end_msg 0 || true
 		else
 			log_end_msg 1 || true
@@ -46,9 +46,9 @@ case "$1" in
 
 	status)
 		if [ -f /var/run/iiod.pid ] ; then
-			status_of_proc -p /var/run/iiod.pid /usr/sbin/iiod iiod && exit 0 || exit $?
+			status_of_proc -p /var/run/iiod.pid @CMAKE_INSTALL_FULL_SBINDIR@/iiod iiod && exit 0 || exit $?
 		else
-			status_of_proc /usr/sbin/iiod iiod && exit 0 || exit $?
+			status_of_proc @CMAKE_INSTALL_FULL_SBINDIR@/iiod iiod && exit 0 || exit $?
 		fi
 		;;
 
-- 
2.11.0

